rpcs3/Emu/RSX/Overlays/overlays.cpp

```diff
@@ -151,49 +151,59 @@ namespace rsx
 					continue;
 				}
 
+				bool ignore_gamepad_input = false;
+
 				// Get keyboard input if supported by the overlay and activated by the game.
 				// Ignored if a keyboard pad handler is active in order to prevent double input.
-				if (m_keyboard_input_enabled && !m_keyboard_pad_handler_active && input::g_keyboards_intercepted)
+				if (m_keyboard_input_enabled)
```

Comment:
This change makes no sense. You're just adding another scope and changing the entire history

Reply:
> This change makes no sense. You're just adding another scope and changing the entire history

Should be fixed now. I was resetting `m_key_repeat` back to false once once `m_keyboard_pad_handler_active` was false, which required me to add another scope, but it looks like I probably didn't have to reset it.

I'm also unsure under what circumstances the following piece of code is supposed to run, because I couldn't get it to trigger when testing MGO.

```cpp
else if (g_cfg.io.keyboard != keyboard_handler::null)
{
	// Workaround if cellKb did not init the keyboard handler.
	handler.Init(1);

	// Enable key repeat
	ensure(!keyboards.empty());
	::at32(keyboards, 0).m_key_repeat = true;
}
```

Reply:
That should happen if you run a game that doesn't use cellKb, as the description says.